### ISSUE TITLE
InlineAssociationDeserializer must provide a type class for StdDeserializer

### ISSUE THREAD

**Issue by hacki11** [BlackPepperSoftware/bowman#7]
A change in Jackson 2.9.0 returns a NullPointerException at
(https://github.com/FasterXML/jackson-databind/issues/1721)

```
InlineAssociationDeserializer(RestOperations restOperations, ClientProxyFactory proxyFactory) {
        this((Class)null, restOperations, proxyFactory);
    }

```
Should be this(Object.class,....)

Unfortunately, the Deserializer can not be extended, due the private constructor and package scoped dependency like RestOperation.

**Comment by hdpe** [comment]
Hi, thanks very much for reporting this!

I've pushed back a fix so that the `InlineAssociationDeserializer` is constructed in a Jackson 2.9.0-compatible way.

This is only half the battle as you then run into the problem with Spring HATEOAS's `Jackson2HalModule` in your linked issue. Fortunately this was fixed in https://github.com/spring-projects/spring-hateoas/issues/583, but this fix has yet to make it into a Spring HATEOAS release.

So... I've pushed a modification to branch [jackson-2.9](https://github.com/BlackPepperSoftware/bowman/tree/jackson-2.9) in this repository, updating the POM to use the latest Spring HATEOAS snapshot (with the fix) and Jackson 2.9. I've added an integration test to verify that the getAll call works correctly with these versions.
